src/merkle/store/mod.rs

```diff
@@ -482,19 +470,22 @@ impl MerkleStore {
     ///
     /// Merges arbitrary values. They may be leafs, nodes, or a mixture of both.
     pub fn merge_roots(&mut self, root1: Word, root2: Word) -> Result<Word, MerkleError> {
-        let root1: RpoDigest = root1.into();
-        let root2: RpoDigest = root2.into();
-
-        let parent: Word = Rpo256::merge(&[root1, root2]).into();
-        self.nodes.insert(
-            parent.into(),
-            Node {
-                left: root1,
-                right: root2,
-            },
-        );
-
-        Ok(parent)
+        let left: RpoDigest = root1.into();
+        let right: RpoDigest = root2.into();
+
+        let parent = Rpo256::merge(&[left, right]);
+        self.nodes.insert(parent, Node { left, right });
+
+        Ok(parent.into())
+    }
+}
+
+// ITERATORS
+// ================================================================================================
+
+impl Extend<InnerNodeInfo> for MerkleStore {
+    fn extend<T: IntoIterator<Item = InnerNodeInfo>>(&mut self, iter: T) {
+        self.add_inner_nodes(iter.into_iter());
     }
 }
```

Comment:
My initial idea was to use the `From` trait, that is not sufficient since that can only be used once. `Extend` better suits the needs of adding multiple structures to the store.

I didn't implement the `IntoIterator` trait for the existing structures because `InnerNodeInfo` is only a partial view of the data (the leaves are no included). Since we can onlyh have one `IntoIterator` per struct it seems better to skip it.

An alternative was to add another trait, say `IntoInnerNodes` and use that. That needs more code and it is more complicated than this implementation.